File: arrays 101/move_zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        
        int length=0, count=0, n;
        n=nums.length;
        
        for(int i=0; i<n; i++)
        {
            if(nums[i]==0)
                count++;
        }
        
        for(int i=0; i<n; i++)
        {
            if (nums[i]!=0)
            {
                nums[length++] = nums[i];
            }
        }
        
        for (int i=(n-1); i>=n-count; i--)
        {
            nums[i] = 0;
        }
    }
}

File: arrays 101/move_zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = 0
        for i in range(len(nums)):
            if nums[i]!=0:
                nums[length] = nums[i]
                length += 1
        
        for i in range(length,len(nums)):
            nums[i] = 0